app/authentication/used_token_storage.py

```diff
@@ -0,0 +1,21 @@
+from sqlalchemy.exc import IntegrityError
+
+from app.data_model.database import UsedToken, commit_or_rollback, db_session
+
+
+def use_token(token):
+    """
+    Use a token
+    :param token: token to mark as used
+    :return: True if the token was used and False if the token has already been used.
+    """
+    try:
+        with commit_or_rollback(db_session):
+            jti = UsedToken(token)
+            # pylint: disable=maybe-no-member
+            # db_session has a add function but it is wrapped in a session_scope which confuses pylint
+            db_session.add(jti)
+    except IntegrityError:
+        return False
+
+    return True
```

Comment:
Might be safer to return False by default and only return True after line 17? Maybe I'm just being paranoid... see later comments about this too